### ISSUE TITLE
Publishing a native AOT library shouldn't publish managed symbols and XML doc files of ProjectReferences

### ISSUE THREAD

**Issue by eerhardt** [dotnet/sdk#40488]
### Describe the bug
When publishing a native AOT library that has a ProjectReference to another class library, the other class library's XML docs and managed .pdb symbols are being published along size the native AOT library. These are unnecessary files and should be excluded by default.

### To Reproduce
Given 2 projects:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <InvariantGlobalization>true</InvariantGlobalization>
    <PublishAot>true</PublishAot>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\AnotherClassLib\AnotherClassLib.csproj" />
  </ItemGroup>
</Project>
```

```C#
using AnotherClassLib;
using System.Runtime.InteropServices;

namespace MyLibrary;

public static class EntryPoints
{
    [UnmanagedCallersOnly(EntryPoint = "MyValidate")]
    public static int MyValidate()
    {
        Class1 c = new();
        c.Method();
        return 1;
    }
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <GenerateDocumentationFile>True</GenerateDocumentationFile>
  </PropertyGroup>

</Project>
```

```C#
namespace AnotherClassLib;

/// <summary>
/// This is Class1.
/// </summary>
public class Class1
{
    /// <summary>
    /// This is a method.
    /// </summary>
    public void Method()
    {
        Console.WriteLine("Hello from a method");
    }
}
```

`dotnet publish -r win-x64` the first project and inspect the publish folder.

![image](https://github.com/dotnet/sdk/assets/8291187/3d471ac2-1909-42b4-8f92-a786e959cdea)

Those `AnotherClassLib.pdb` and `AnotherClassLib.xml` files shouldn't be there. They are unnecessary.

### Exceptions (if any)
N/A

cc @agocke @MichalStrehovsky @sbomer

**Comment by MichalStrehovsky** [comment]
The AnotherClassLib.xml is there because SDK says that we publish those XMLs by default. It's not specific to building native AOT libraries, it's for any kind of publish.

You can opt out with `PublishDocumentationFiles=false`. I don't understand why we need to publish these _in general_. You might have more context as you signed off on it in #1062. It might be better to separate this part of the problem into a separate issue so that it can be evaluated under multiple angles (any implications on PublishTrimmed or PublishSingleFile?).

The PDB of the reference ends up in the output because we don't exclude it in ILCompiler targets. I think we should really do the proper fix discussed at https://github.com/dotnet/runtime/pull/98342 to fix this. We don't have an issue tracking that. This can be the issue.

**Comment by eerhardt** [comment]
> The AnotherClassLib.xml is there because SDK says that we publish those XMLs by default. It's not specific to building native AOT libraries, it's for any kind of publish.

Good call. I forgot that we did that. I believe its main reason was for things like swagger / OpenAPI runtime document generation. But I agree that in general non-ASP.NET apps shouldn't need this behavior.

> The PDB of the reference ends up in the output because we don't exclude it in ILCompiler targets. I think we should really do the proper fix discussed at https://github.com/dotnet/runtime/pull/98342 to fix this. We don't have an issue tracking that. This can be the issue.

I've renamed and updated the top comment to just be about managed symbols.
